src/mantine-form/src/resolvers/zod-resolver/zod-resolver.ts

```diff
@@ -5,30 +5,32 @@ interface ZodError {
   message: string;
 }
 
-interface ZodResults {
-  success: boolean;
+interface ZodParseSuccess {
+  success: true;
+}
+
+interface ZodParseError {
+  success: false;
   error: {
     errors: ZodError[];
   };
 }
 
-interface ZodSchema {
-  safeParse(values: Record<string, any>): ZodResults;
+interface ZodSchema<T extends Record<string, any>> {
+  safeParse(values: T): ZodParseSuccess | ZodParseError;
 }
 
-export function zodResolver<T extends Record<string, any>>(schema: any) {
-  const _schema: ZodSchema = schema;
-
+export function zodResolver<T extends Record<string, any>>(schema: ZodSchema<T>) {
   return (values: T): FormErrors => {
-    const parsed = _schema.safeParse(values);
+    const parsed = schema.safeParse(values);
 
     if (parsed.success) {
       return {};
     }
 
     const results = {};
 
-    parsed.error.errors.forEach((error) => {
+    (parsed as ZodParseError).error.errors.forEach((error) => {
```

Comment:
Required, not sure why TS don't infer this

Reply:
You can try narrowing the type using the `error` property, e.g. `const isError = 'error' in parsed` and then `if (!isError)` should do the trick.